cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -339,11 +339,13 @@ else if (clearSnapshotStrategy.hasTTL())
     private CompletionStage<Map<String, AvailabilityHint>> createSnapshot(ClientConfig options, RingResponse ring)
     {
         Map<String, PartitionedDataLayer.AvailabilityHint> availabilityHints = new ConcurrentHashMap<>(ring.size());
+        Map<String, Boolean> distinctInstances = new HashMap<>();
 
         // Fire off create snapshot request across the entire cluster
         List<CompletableFuture<Void>> futures =
         ring.stream()
             .filter(ringEntry -> datacenter == null || datacenter.equals(ringEntry.datacenter()))
+            .filter(ringEntry -> !options.createSnapshotFilterDistinctInstances() || distinctInstances.putIfAbsent(ringEntry.fqdn(), true) == null)
```

Comment:
NIT, we can alternatively use a HashSet
```suggestion
            .filter(ringEntry -> !options.createSnapshotFilterDistinctInstances() || distinctInstances.add(ringEntry.fqdn()))
```